geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -57,18 +83,34 @@ public ReturnCode filterKeyValue(Cell v) throws IOException {
         if (filter == null) {
             return ReturnCode.INCLUDE;
         } else {
-            SimpleFeature sf = serializer.deserialize(v.getValueArray(), v.getValueOffset(), v.getValueLength());
-            if (filter.evaluate(sf)) {
-                return ReturnCode.INCLUDE;
-            } else {
+            try {
+                log.trace("cell = {}", v);
+                byte[] valueBytes = v.getValueArray();
+                int valueOffset = v.getValueOffset();
+                int valueLength = v.getValueLength();
+                log.trace("valueBytes.length = {}, valueOffset = {}, valueLength = {}, column = {}",
+                        valueBytes.length,
+                        valueOffset,
+                        valueLength,
+                        Bytes.toString(v.getFamilyArray(), v.getFamilyOffset(), v.getFamilyLength())
+                );
```

Comment:
I'd argue that this block should do one of ...
1.  Not exist since it is debugging code
2.  Be blocked off by a log.ifTraceEnabled)
3.  Reuse the variables set up in the next line.